### PR TITLE
[Keyboard] Dumbpad v1x_oled

### PR DESCRIPTION
**Pull request by medoix**
## Description

This PR adds the QMK config files for the Dumbpad v1x_oled variant

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dumbpad/v1x_oled/config.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#define MATRIX_COLS 5
+
+#define MATRIX_ROW_PINS { F4, F5, F6, F7 }
+#define MATRIX_COL_PINS { C6, B1, B3, B2, B6 }
+#define UNUSED_PINS
+
+/* Dual rotary encoders */
+#define ENCODERS_PAD_A { B5, D7 }
+#define ENCODERS_PAD_B { B4, E6 }
+#define ENCODER_DIRECTION_FLIP
```
Comment (zvecr) [inline keyboards/dumbpad/v1x_oled/config.h:34]:
```suggestion
#define ENCODERS_PAD_A { B4, E6 }
#define ENCODERS_PAD_B { B5, D7 }
```

File: keyboards/dumbpad/v1x_oled/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
+
+EXTRAFLAGS += -flto
```
Comment (zvecr) [inline keyboards/dumbpad/v1x_oled/keymaps/via/rules.mk:3]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+#include <stdio.h>
+char wpm_str[10];
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c:19]:
```suggestion
```

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    } else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_MNXT);
+        } else {
+            tap_code(KC_MPRV);
+        }
+    }
+    return true;
+}
+#endif
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c:93]:
```suggestion
```

File: keyboards/dumbpad/v1x_oled/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    // Left encoder
+    } else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/keymaps/via/keymap.c:92]:
```suggestion
```

File: keyboards/dumbpad/v1x_oled/v1x_oled.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "v1x_oled.h"
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/v1x_oled.c:16]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }
    // Right encoder
    if (index == 0) {
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code(KC_VOLD);
        }
    // Left encoder
    } else if (index == 1) {
        if (clockwise) {
            tap_code(KC_MNXT);
        } else {
            tap_code(KC_MPRV);
        }
    }
    return true;
}
#endif
```

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+
+#ifdef OLED_ENABLE
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c:95]: ALso, this should probably be moved to the `v1x_oled.c` file, like with the encoder.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c (hunk 3 above)
Comment (medoix) [reply keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c:19]: This is needed for the ```oled_task_user()``` function

Comment (medoix) [reply keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c]: Moved out of keymap in https://github.com/qmk/qmk_firmware/pull/17058/commits/165318675c78d8f4697645e237d2aff5672aa12d

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c (hunk 4 above)
Comment (medoix) [reply keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c]: Moved out of keymap in https://github.com/qmk/qmk_firmware/pull/17058/commits/165318675c78d8f4697645e237d2aff5672aa12d

File: keyboards/dumbpad/v1x_oled/keymaps/via/keymap.c (hunk 5 above)
Comment (medoix) [reply keyboards/dumbpad/v1x_oled/keymaps/via/keymap.c]: Moved out of keymap in https://github.com/qmk/qmk_firmware/pull/17058/commits/165318675c78d8f4697645e237d2aff5672aa12d

File: keyboards/dumbpad/v1x_oled/v1x_oled.c (hunk 6 above)
Comment (medoix) [reply keyboards/dumbpad/v1x_oled/v1x_oled.c:16]: As suggested i have moved the ENCODER code out of the keymap files and into the v1x_oled.c file in https://github.com/qmk/qmk_firmware/pull/17058/commits/165318675c78d8f4697645e237d2aff5672aa12d

File: keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c (hunk 7 above)
Comment (medoix) [reply keyboards/dumbpad/v1x_oled/keymaps/default/keymap.c]: Moved OLED code out of keymap files in https://github.com/qmk/qmk_firmware/pull/17058/commits/165318675c78d8f4697645e237d2aff5672aa12d

File: keyboards/dumbpad/v1x_oled/rules.mk [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/rules.mk:12]:
```suggestion
```

File: keyboards/dumbpad/v1x_oled/v1x_oled.c [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        }
+    }
+}
+
+// Used to draw on to the oled screen
+bool oled_task_user(void) {
+    render_anim();  // renders pixelart
+
+    oled_set_cursor(0, 0);                            // sets cursor to (row, column) using charactar spacing (5 rows on 128x32 screen, anything more will overflow back to the top)
+    sprintf(wpm_str, "WPM:%03d", get_current_wpm());  // edit the string to change what shows up, edit %03d to change how many digits show up
```
Comment (drashna) [inline keyboards/dumbpad/v1x_oled/v1x_oled.c:145]:
```suggestion
    oled_write_P(PSTR("WPM: "), false);
    oled_write(get_u8_str(get_current_wpm(), '0'), false);
```

**Review by drashna — APPROVED** [review]

**Comment by medoix** [comment]
@zvecr any additional feedback or can we please merge this in?

**Comment by medoix** [comment]
bump, can someone please review :pray:

File: keyboards/dumbpad/v1x_oled/rules.mk [hunk 10]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_oled/rules.mk:7]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/dumbpad/v1x_oled/rules.mk [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/dumbpad/v1x_oled/rules.mk:13]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**